### DISCUSSION TITLE
Custom log file names

### DISCUSSION THREAD

**Discussion by resucutie** [ThioJoe/YT-Spammer-Purge#414] [Ideas]
It is hard to manage a big list of log files with them named "SpamLog-*numbers and letters*". So my idea is to introduce to customizable names. My idea is to add a setting called "log_name" that will allow you to put custom names in them as how much as you want. There will also be included custom props:
* `{channel}`: The name of the channel
* `{video}`: The name of the scanned video
* `{videos_list}`: A list of all scanned video names (if scanned multiple)
* `{videos_counter}`: A count of all scanned video names
* `{date}`: The date that the log file was created
* `{time}`: The time that the log file was created

**Comment by TechStudent10** [comment]
I support this

**Comment by ThioJoe** [comment]
Hm yea this could probably be done

**Comment by Firecul** [comment]
`{videos_list}` would probably get very unusable very quickly. 
> In the Windows API [...] the maximum length for a path is MAX_PATH, which is defined as 260 characters.
https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd

File names also have a limit of around 255 in most file systems (or OSes anyway)

**Reply by resucutie** [reply]
good point

**Comment by Firecul** [comment]
What about a modification of the idea, a log folder structure? {Channel Name} > Logs with your suggested optional names e.g. `{video}-{date}-{time}.txt`

**Reply by TechStudent10** [reply]
Also a good idea!

**Reply by resucutie** [reply]
Maybe `{video} at {date} {time}.txt`. Usually people uses spaces instead of dashes

**Reply by Firecul** [reply]
One thing I thought about as an extention to this idea would be to save it like that with `{channel} > {video id} {date} {time}.txt` and the program could check last scanned date and (configurably) skip if it's within a certain time frame to save API calls. Best used when scanning multiple videos from a channel.
